packages/protocol/test/governance/network/governance.ts

```diff
@@ -1500,6 +1500,113 @@ contract('Governance', (accounts: string[]) => {
+      assert.isTrue(success)
+    })
+
+    it("should update the voter's vote record", async () => {
+      await governance.revokeVotes()
+      const [recordProposalId, recordValue, recordWeight] = await governance.getVoteRecord(
+        account,
+        index
+      )
+      assertEqualBN(recordProposalId, proposalId)
+      assertEqualBN(recordValue, VoteValue.None)
+      assertEqualBN(recordWeight, 0)
+    })
+
+    it('should unset the most recent referendum proposal voted on', async () => {
+      await governance.revokeVotes()
+      assert.equal((await governance.getMostRecentReferendumProposal(account)).toNumber(), 0)
+    })
+
+    it('should return false on `isVoting`', async () => {
+      await governance.revokeVotes()
+      const voting = await governance.isVoting(accounts[0])
+      assert.isFalse(voting)
+    })
+
+    it('should emit the AccountProposalsVoteRevoked event', async () => {
+      const resp = await governance.revokeVotes()
+      assert.equal(resp.logs.length, 1)
+      const log = resp.logs[0]
+      assertLogMatches2(log, {
+        event: 'AccountProposalsVoteRevoked',
+        args: {
+          account,
+        },
+      })
+    })
+
+    describe('when voting on two proposals', () => {
+      const proposalId1 = 2
+      const proposalId2 = 3
+      const index1 = 1
+      const index2 = 2
+      beforeEach(async () => {
+        const newDequeueFrequency = 60
+        await governance.setDequeueFrequency(newDequeueFrequency)
+        await governance.propose(
+          [transactionSuccess1.value],
+          [transactionSuccess1.destination],
+          // @ts-ignore bytes type
+          transactionSuccess1.data,
+          [transactionSuccess1.data.length],
+          descriptionUrl,
+          // @ts-ignore: TODO(mcortesi) fix typings for TransactionDetails
+          { value: minDeposit }
+        )
+        await timeTravel(newDequeueFrequency, web3)
+        await governance.approve(proposalId1, index1)
+        await governance.propose(
+          [transactionSuccess2.value],
+          [transactionSuccess2.destination],
+          // @ts-ignore bytes type
+          transactionSuccess2.data,
+          [transactionSuccess2.data.length],
+          descriptionUrl,
+          // @ts-ignore: TODO(mcortesi) fix typings for TransactionDetails
+          { value: minDeposit }
+        )
+        await timeTravel(newDequeueFrequency, web3)
+        await governance.approve(proposalId2, index2)
+        await timeTravel(approvalStageDuration, web3)
+        await mockLockedGold.setAccountTotalLockedGold(account, weight)
+        await governance.vote(proposalId2, index2, value)
+        await governance.vote(proposalId1, index1, value)
+      })
+
+      it('should return false on `isVoting`', async () => {
+        await governance.revokeVotes()
+        const voting = await governance.isVoting(accounts[0])
+        assert.isFalse(voting)
+      })
+    })
+  })
```

Comment:
if we emit events for each unvoted proposal, would love an additional unit test here for the two proposals `describe`